### PR TITLE
Added TCS parameters file path in consist files (+ refactoring and code cleanup)

### PR DESCRIPTION
**Pull request by Sharpe49**
https://blueprints.launchpad.net/or/+spec/consist-tcs-parameters

Changes and refactoring of the rolling stock loading code was needed to give access to the Train object during the loading of the TrainControlSystem object of the locomotives.

### CODE REVIEW

**Comment by Sharpe49** [comment]
Ok, the merge bot doesn't like the fact that I replaced all tabs by spaces in MPManager.cs...

**Comment by Csantucci** [comment]
Please have a look at this thread http://www.elvastower.com/forums/index.php?/topic/35945-disrupted-train-after-restore-in-u20220126-1022/

**Comment by Sharpe49** [comment]
I also got a report from one of my friends on my custom version of OR about another issue which may be related too: in timetable mode, some cars are getting flipped wrongly.
I am investigating both issues.

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/TrainControlSystem.cs [hunk 1]
```diff
@@ -861,18 +868,35 @@ public void HandleEvent(PowerSupplyEvent evt, string message)
...
+            {
+                buffer = new string('\0', 256);
+                length = NativeMethods.GetPrivateProfileString(sectionName, keyName, null, buffer, buffer.Length, ParametersFileName);
+
+                if (length > 0)
+                {
+                    buffer.Trim();
+                    return (T)Convert.ChangeType(buffer, typeof(T), System.Globalization.CultureInfo.InvariantCulture);
+                }
+            }
```
[the remark this answers is not in the document]
Comment (Sharpe49) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/TrainControlSystem.cs:896]: I don't know... it won't be used anywhere else.

[the remark this answers is not in the document]
Comment (cesarBLG) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/TrainControlSystem.cs:896]: OK, let's keep it as is then.

**Review by cesarBLG — APPROVED** [review]
This has been in unstable for a while now, so I suppose that there are no more bugs due to adding cars twice. The code seems good.

**Comment by Sharpe49** [comment]
Thank you.
Since the blueprint has been approved, I am merging this.

**Pull request merged**